### DISCUSSION TITLE
There is a way to auto increment buildNumber/versionCode?

### DISCUSSION THREAD

**Discussion by gabsdotco** [expo/expo#17606] [Archived: Questions] [answered]
Hello! I'm building a expo app using expo-dev-client and I need to publish the app to the store. The problem is, everytime I make a publish to the store an error occurs saying that I need to increment the versionCode/buildNumber because a binary already have that code published on the store and I understand that, but there is a way to auto increment that? 

Like, I want to create a publish action on github and always having to increment manually is not cool.

**Comment by b0iq** [comment, marked as answer]
@kbrandwijk says:
> This feature is available on both platforms for EAS builds through the autoIncrement field in eas.json. For more information on how to configure this, check out the documentation [here](https://docs.expo.dev/build-reference/eas-json/).

**Reply by gabsdotco** [reply]
Thank you so much for this!

**Reply by calderwoodra** [reply]
Seems like this doesn't cover `expo.version` from what I can tell, which the apple app store will block if a duplicate is uploaded.

```
yarn run v1.22.19
$ eas build -p ios --profile production --auto-submit --no-wait
    InvalidEasJsonError: eas.json is not valid.
    - "build.production.autoIncrement" must be a boolean
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

<img width="1071" alt="Screen Shot 2022-10-18 at 2 22 25 AM" src="https://user-images.githubusercontent.com/6539032/196391232-df5f7a76-7117-4ad7-94e8-9d58c357f712.png">

**Reply by brentvatne** [reply]
@calderwoodra - that is correct. we expect that users typically want to bump the version on their own since that is visible to users and is often used to communicate some meaning. we may add support for auto incrementing that in the future though! for now we just handle buildNumber and versionCode

**Comment by konjoinfinity** [comment]
It’s not obvious in the docs, it must be under the build section in eas.json:

  "build": {
    "development": {
      "developmentClient": true,
      "distribution": "internal",
      "channel": "development",
      "autoIncrement": true
    },

**Reply by stesvis** [reply]
Does this work also with `eas update`?

**Comment by jagdish1o1** [comment]
Dropping this link for more details. 

App version management with Expo CLI

https://docs.expo.dev/build-reference/app-versions/

**Comment by boblitex** [comment]
is there a way to achieve this even while having a dynamic config (app.config.js)

**Reply by brentvatne** [reply]
yup, use the remote version source: https://docs.expo.dev/build-reference/app-versions/#remote-version-source

it's recommended in most cases anyhow

**Reply by prometx11** [reply]
Is there a way to use remote version source when using `eas build --local` ?

**Reply by brentvatne** [reply]
its not supported at the moment, the `local` flag isn't intended to support all `remote` features (as the name implies). you could set your build number based on the result of `eas build:version:get`, eg: `eas build:version:get -p ios --non-interactive --json` and then parse the json

**Reply by mlazari** [reply]
@brentvatne I use `"appVersionSource": "remote"`. For profiles that have `"autoIncrement": true` it increments and uses the new version just as I expect, but for the ones that don't have that it just always uses 1 for the version. Is that expected or I am doing something wrong? My expectation is that it would use the remote version without incrementing it in that case.

L.E. I think it works as I expect and I see 1 because I use a different bundle id for the profile without autoincrement.
